dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import unicode_literals
+
+import os
+import posixpath
+
+from funcy import cached_property
+from ratelimit import limits, sleep_and_retry
+from backoff import on_exception, expo
+
+from dvc.scheme import Schemes
+from dvc.path_info import CloudURLInfo
+from dvc.remote.base import RemoteBASE
+from dvc.config import Config
+from dvc.exceptions import DvcException
+from dvc.remote.gdrive.pydrive import (
+    RequestListFile,
+    RequestListFilePaginated,
+    RequestUploadFile,
+    RequestDownloadFile,
+)
+from dvc.remote.gdrive.utils import FOLDER_MIME_TYPE
+
+
+class GDriveURLInfo(CloudURLInfo):
+    @property
+    def netloc(self):
+        return self.parsed.netloc
+
+
+class RemoteGDrive(RemoteBASE):
+    scheme = Schemes.GDRIVE
+    path_cls = GDriveURLInfo
+    REGEX = r"^gdrive://.*$"
+    REQUIRES = {"pydrive": "pydrive"}
+
+    def __init__(self, repo, config):
+        super(RemoteGDrive, self).__init__(repo, config)
+        self.no_traverse = False
+        self.cached_dirs = {}
+        self.cached_ids = {}
+        self.path_info = self.path_cls(config[Config.SECTION_REMOTE_URL])
+        self.config = config
+        self.init_drive()
+
+    def init_drive(self):
+        if Config.SECTION_REMOTE_KEY_FILE not in self.config:
+            raise DvcException(
+                "Google Drive settings file path is missed from config. "
+                "Learn more at https://dvc.org/doc."
```

Comment:
better put a link to the `remote/add` or `remote/modify` directly?

Reply:
should we actually move this check and raise an exception down the path when we try to read it and fail?